Q1: Write a python program to scrape data for “Data Analyst” Job position in “Bangalore” location. You have to scrape the job-title, job-location, company_name, experience_required. You have to scrape first 10 jobs data.
This task will be done in following steps:
1. First get the webpage https://www.shine.com/
2. Enter “Data Analyst” in “Job title, Skills” field and enter “Bangalore” in “enter the location” field.
3. Then click the search button.
4. Then scrape the data for the first 10 jobs results you get.
5. Finally create a dataframe of the scraped data.

In [1]:
# Importing Necessary Libraries
import selenium
from selenium import webdriver
import pandas as pd
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
import warnings
warnings.filterwarnings("ignore")
import time

In [2]:
# Opening an automated web browser and website
driver = webdriver.Chrome()
driver.get("https://www.shine.com/")

In [4]:
# Scraping the Job field and entering "Data Analyst" in the field
job_field = driver.find_element(By.ID,"id_q")
job_field.send_keys("Data Analyst")

In [5]:
# Scraping location field and entering "Bangalore" in the empty field.
job_location = driver.find_element(By.ID,"id_loc")
job_location.send_keys("Bangalore")

In [6]:
search = driver.find_element(By.XPATH,"/html/body/div/div[4]/div/div[2]/div[2]/div/form/div/div[2]/div/button")
search.click()

In [31]:
# Scraping title of the job profile
title_tags = driver.find_elements(By.XPATH,'//h2[@itemprop="name"]')
title_tags

[<selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4376")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4467")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4468")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4469")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4470")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4471")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b

In [32]:
# Creating an empty list for job titles and looping through the first 10 results.
job_titles = []
for i in title_tags:
    if i.text is None:
        job_titles.append('Not')
    else:
        job_titles.append(i.text)

job_titles[:10]

['Data Analyst',
 'Vacancy For Data Analyst',
 'Project Coordinator (Data Analyst) Bangalore',
 'Project Coordinator (Data analyst) Bangalore',
 'Data analyst - Java / Python',
 'Business Data Analyst - Modeling & Profiling',
 'Data Analyst - Java/Python BANGALORE',
 'Data Analyst - Java/Python',
 'Data Analyst - SQL/ Database Maintenance',
 'Business Data Analyst - Modeling & Profiling']

In [34]:
# Scraping location of the job profile
location_tags = driver.find_elements(By.XPATH,'//div[@class=" jobCard_jobCard_lists_item__YxRkV jobCard_locationIcon__zrWt2"]')
location_tags

[<selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4715")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4717")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4718")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4719")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4706")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4720")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b

In [35]:
# Creating an empty list for job titles and looping through the first 10 results.
location = []
for i in location_tags:
    location.append(i.text)
    
location[:10]

['Bangalore\n+9',
 'Bangalore\n+13',
 'Bangalore',
 'Bangalore',
 'Bangalore',
 'Bangalore',
 'Bangalore',
 'Bangalore',
 'Bangalore',
 'Bangalore']

In [36]:
# Scraping company name of the job profile
company_tags = driver.find_elements(By.XPATH,'//div[@class="jobCard_jobCard_cName__mYnow"]')
company_tags

[<selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4377")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4682")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4691")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4701")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4670")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4894")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b

In [37]:
# Creating an empty list for job titles and looping through the first 10 results.
company_name = []
for i in company_tags:
    company_name.append(i.text)
    
company_name[:10]

['hire connect',
 'kavya interprises',
 'futures and careers',
 'futures and careers',
 'boyen haddin consulting and technol...',
 'boyen haddin consulting and technol...',
 'boyen haddin consulting and technol...',
 'boyen haddin consulting and technol...',
 'boyen haddin consulting and technol...',
 'boyen haddin consulting and technol...']

In [38]:
# Scraping experience required for the job profile
exp_tags = driver.find_elements(By.XPATH,'//div[@class=" jobCard_jobCard_lists_item__YxRkV jobCard_jobIcon__3FB1t"]')
exp_tags

[<selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4713")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4910")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4911")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4912")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4707")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b053413af426510b6", element="D1115E76FDF4E516FE24EC2E41BA431B_element_4913")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fb066113abb0784b

In [39]:
# Creating an empty list for job titles and looping through the first 10 results.
exp_required = []
for i in exp_tags:
    exp_required.append(i.text)
    
exp_required[:10]

['0 to 1 Yr',
 '0 to 4 Yrs',
 '2 to 4 Yrs',
 '2 to 4 Yrs',
 '5 to 10 Yrs',
 '2 to 6 Yrs',
 '4 to 8 Yrs',
 '2 to 6 Yrs',
 '2 to 6 Yrs',
 '2 to 6 Yrs']

In [41]:
# Printing length to check if all the columns match or not.
print(len(exp_required[:10]),len(company_name[:10]),len(location[:10]),len(job_title[:10]))

10 10 10 10


In [43]:
# Creating a pandas dataframe to store all the scraped data.
jobs = pd.DataFrame({})
jobs["Title"] = job_titles[:10]
jobs["Location"] = location[:10]
jobs["Experience Required"] = exp_required[:10]
jobs["Company"] = company_name[:10]

In [44]:
jobs

Title        Location  \
0                                  Data Analyst   Bangalore\n+9   
1                      Vacancy For Data Analyst  Bangalore\n+13   
2  Project Coordinator (Data Analyst) Bangalore       Bangalore   
3  Project Coordinator (Data analyst) Bangalore       Bangalore   
4                  Data analyst - Java / Python       Bangalore   
5  Business Data Analyst - Modeling & Profiling       Bangalore   
6          Data Analyst - Java/Python BANGALORE       Bangalore   
7                    Data Analyst - Java/Python       Bangalore   
8      Data Analyst - SQL/ Database Maintenance       Bangalore   
9  Business Data Analyst - Modeling & Profiling       Bangalore   

  Experience Required                                 Company  
0           0 to 1 Yr                            hire connect  
1          0 to 4 Yrs                       kavya interprises  
2          2 to 4 Yrs                     futures and careers  
3          2 to 4 Yrs                     futures and careers  
4         5 to 10 Yrs  boyen haddin consulting and technol...  
5          2 to 6 Yrs  boyen haddin consulting and technol...  
6          4 to 8 Yrs  boyen haddin consulting and technol...  
7          2 to 6 Yrs  boyen haddin consulting and technol...  
8          2 to 6 Yrs  boyen haddin consulting and technol...  
9          2 to 6 Yrs  boyen haddin consulting and technol...

Q2: Write a python program to scrape data for “Data Scientist” Job position in “Bangalore” location. You have to scrape the job-title, job-location, company_name. You have to scrape first 10 jobs data.
This task will be done in following steps:
1. First get the webpage https://www.shine.com/
2. Enter “Data Scientist” in “Job title, Skills” field and enter “Bangalore” in “enter the location” field.
3. Then click the search button.
4. Then scrape the data for the first 10 jobs results you get.
5. Finally create a dataframe of the scraped data.

In [46]:
# Opening an automated web browser and website
driver = webdriver.Chrome()
driver.get('https://www.shine.com/')

In [47]:
# Scraping the Job field and entering "Data Scientist" in the field
designation1 = driver.find_element(By.ID,'id_q')
designation1.send_keys('Data Scientist')

In [48]:
# Scraping the Location field and entering "Bangalore" in the field
location1 = driver.find_element(By.ID,'id_loc')
location1.send_keys('Bangalore')

In [49]:
search1 = driver.find_element(By.XPATH,'/html/body/div/div[4]/div/div[2]/div[2]/div/form/div/div[2]/div/button')
search1.click()

In [50]:
# Scraping title of the job profile
title_tags1 = driver.find_elements(By.XPATH,'//h2[@itemprop="name"]')
title_tags1

[<selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2550")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2556")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2557")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2558")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2559")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2560")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab6

In [52]:
# Creating an empty list for job titles and looping through the first 10 results.
title_name1 = []
for i in title_tags1:
    if i.text is None:
        title_name1.append('Not')
    else:
        title_name1.append(i.text)
        
title_name1[:10]

['Data Scientist - Business Intelligence',
 'Data Scientist for Bangalore',
 'Hiring For Data Scientist',
 'Hiring For Data Scientist',
 'Data Scientist - Machine Learning',
 'Data Scientist - Spark/Python/Redshift',
 'Data Scientist',
 'Data Scientist- Associate',
 'Data Scientist Vacancy',
 'Data Scientist - Business Intelligence']

In [53]:
# Scraping location of the job profile
location1 = driver.find_elements(By.XPATH,'//div[@class=" jobCard_jobCard_lists_item__YxRkV jobCard_locationIcon__zrWt2"]')
location1

[<selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2577")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2584")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2585")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2586")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2587")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2588")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab6

In [56]:
# Creating an empty list for job location and looping through the first 10 results.
location_name1 = []
for i in location1:
    location_name1.append(i.text)
    
location_name1[:10]

['Bangalore',
 'Bangalore',
 'Bangalore\n+13',
 'Bangalore\n+13',
 'Bangalore',
 'Bangalore',
 'Bangalore',
 'Bangalore',
 'Bangalore\n+14',
 'Bangalore']

In [57]:
# Scraping the experience required detail
exp_given = driver.find_elements(By.XPATH,'//div[@class=" jobCard_jobCard_lists_item__YxRkV jobCard_jobIcon__3FB1t"]')
exp_given

[<selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2578")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2603")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2604")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2605")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2606")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2607")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab6

In [58]:
# Creating an empty list for experience required and looping through the first 10 results.
exp_req = []
for i in exp_given:
    exp_req.append(i.text)
    
exp_req[:10]

['2 to 6 Yrs',
 '0 to 3 Yrs',
 '0 to 4 Yrs',
 '0 to 4 Yrs',
 '2 to 4 Yrs',
 '4 to 8 Yrs',
 '10 to 12 Yrs',
 '4 to 6 Yrs',
 '0 to 4 Yrs',
 '2 to 6 Yrs']

In [59]:
# Scraping the Company name details from the website
company_tags1 = driver.find_elements(By.XPATH,'//div[@class="jobCard_jobCard_cName__mYnow"]')
company_tags1

[<selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2551")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2624")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2625")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2626")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2627")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab61ee29428d6c39d8c", element="47C6F445DD57B54869A46AC7D592B15A_element_2628")>,
 <selenium.webdriver.remote.webelement.WebElement (session="57b8814cfd5d6ab6

In [61]:
# Creating an empty list for company names and looping through the first 10 results.
company_name1 = []
for i in company_tags1:
    company_name1.append(i.text)

company_name1[:10]

['boyen haddin consulting and technol...',
 'get hired',
 'kavya interprises',
 'kavya interprises',
 'boyen haddin consulting and technol...',
 'boyen haddin consulting and technol...',
 'ibm india pvt. limited',
 'jpmorgan',
 'divya interprises',
 'boyen haddin consulting and technol...']

In [64]:
# Creating a Pandas Dataframe
jobs1 = pd.DataFrame({})
jobs1["Job Title"] = title_name1[:10]
jobs1["Company Name"] = company_name1[:10]
jobs1["Experience Required"] = exp_req[:10]
jobs1["Location"] = location_name1[:10]

In [65]:
jobs1

Job Title  \
0  Data Scientist - Business Intelligence   
1            Data Scientist for Bangalore   
2               Hiring For Data Scientist   
3               Hiring For Data Scientist   
4       Data Scientist - Machine Learning   
5  Data Scientist - Spark/Python/Redshift   
6                          Data Scientist   
7               Data Scientist- Associate   
8                  Data Scientist Vacancy   
9  Data Scientist - Business Intelligence   

                             Company Name Experience Required        Location  
0  boyen haddin consulting and technol...          2 to 6 Yrs       Bangalore  
1                               get hired          0 to 3 Yrs       Bangalore  
2                       kavya interprises          0 to 4 Yrs  Bangalore\n+13  
3                       kavya interprises          0 to 4 Yrs  Bangalore\n+13  
4  boyen haddin consulting and technol...          2 to 4 Yrs       Bangalore  
5  boyen haddin consulting and technol...          4 to 8 Yrs       Bangalore  
6                  ibm india pvt. limited        10 to 12 Yrs       Bangalore  
7                                jpmorgan          4 to 6 Yrs       Bangalore  
8                       divya interprises          0 to 4 Yrs  Bangalore\n+14  
9  boyen haddin consulting and technol...          2 to 6 Yrs       Bangalore

Q3: In this question you have to scrape data using the filters available on the webpage
You have to use the location and salary filter.
You have to scrape data for “Data Scientist” designation for first 10 job results.
You have to scrape the job-title, job-location, company name, experience required. The location filter to be used is “Delhi/NCR”. The salary filter to be used is “3-6” lakhs
The task will be done as shown in the below steps:
1. first get the web page https://www.shine.com/
2. Enter “Data Scientist” in “Skill, Designations, and Companies” field.
3. Then click the search button.
4. Then apply the location filter and salary filter by checking the respective boxes
5. Then scrape the data for the first 10 jobs results you get.
6. Finally create a dataframe of the scraped data.

In [66]:
# Opening an automated web browser and website
driver = webdriver.Chrome()
driver.get("https://www.shine.com/")

In [68]:
# Scraping the Job field and entering "Data Scientist" in the field
designation2 = driver.find_element(By.ID,'id_q')
designation2.send_keys('Data Scientist')

In [69]:
search2 = driver.find_element(By.XPATH,'/html/body/div/div[4]/div/div[2]/div[2]/div/form/div/div[2]/div/button')
search2.click()

In [83]:
# Applying the Location filter 
location_filter = driver.find_element(By.XPATH,'//button[@class="filter_filterBtn__FAqUy btn btn-outline-grey font-size-xs "]')
location_filter.click()

In [84]:
# Selecting the Location Filter and selecting Delhi on the filter
loca_filter_search = driver.find_element(By.ID,'search')
loca_filter_search.send_keys('Delhi')
show_results = driver.find_element(By.XPATH,'//body[@class="styledcls_body_hidden"]')
show_results.click()

In [91]:
# Applying a Salary FIlter
salary_filter = driver.find_element(By.CLASS_NAME,'filter_filter_lists_items__wlFfo')
salary_filter.click()

In [94]:
# Scraping title of the job profile
title_tag2 = driver.find_elements(By.XPATH,'//h2[@itemprop="name"]')
title_tag2

[<selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11922")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11927")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11928")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11929")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11930")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11931")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d46

In [97]:
# Creating an empty list for job titles and looping through the first 10 results.
title_job_name = []
for i in title_tag2:
    title_job_name.append(i.text)
    
title_job_name[:10]

['Data Scientist for Delhi',
 'Data Scientist- Delhi',
 'Data Scientist-Delhi',
 'Clinical Data Management',
 'Data Scientist- Delhi',
 'Data Scientist- Delhi',
 'Clinical Data Manager',
 'Work Full/part-time / Work From Home / Computer Operato ...',
 'Typing Work-Data Entry-Computer Operator-Freelancing-Wo ...',
 'Clinical Data Analyst']

In [98]:
# Scraping location of the job profile
location_tag2 = driver.find_elements(By.XPATH,'//div[@class=" jobCard_jobCard_lists_item__YxRkV jobCard_locationIcon__zrWt2"]')
location_tag2

[<selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_2064")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11950")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11951")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11952")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11953")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11954")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d461

In [104]:
# Creating an empty list for job location and looping through the first 10 results.
job_location2 = []
for i in location_tag2:
    job_location2.append(i.text.replace('\n+',''))
    
job_location2[:10]

['Delhi',
 'Delhi',
 'Delhi',
 'Delhi6',
 'Delhi',
 'Delhi',
 'Delhi6',
 'Delhi4',
 'Delhi4',
 'Delhi6']

In [105]:
# Scraping the Company name details from the website
company_tag2 = driver.find_elements(By.XPATH,'//div[@class="jobCard_jobCard_cName__mYnow"]')
company_tag2

[<selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11923")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11985")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11986")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11987")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11988")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_11989")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d46

In [107]:
# Creating an empty list for company names and looping through the first 10 results.
company_name2 = []
for i in company_tag2:
    company_name2.append(i.text)
    

company_name2[:10]

['get hired',
 'the fashion cosmo',
 'shiva hr services',
 'techno endura',
 'the fashion cosmo',
 'the fashion cosmo',
 'techno endura',
 'desnor destination pvt. ltd.',
 'desnor destination pvt. ltd.',
 'techno endura']

In [108]:
# Scraping the experience required detail
exp_tag2 = driver.find_elements(By.XPATH,'//div[@class=" jobCard_jobCard_lists_item__YxRkV jobCard_jobIcon__3FB1t"]')
exp_tag2

[<selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_2063")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_12004")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_12005")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_12006")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_12007")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d4617917af6d91746074a8283", element="BD43A3D0DFAEAA6D410263B6244C0931_element_12008")>,
 <selenium.webdriver.remote.webelement.WebElement (session="293dd49d461

In [112]:
# Creating an empty list for experience required and looping through the first 10 results.
exp_required2 = []
for i in exp_tag2:
    exp_required2.append(i.text)
    
exp_required2[:10]

['0 to 3 Yrs',
 '0 to 3 Yrs',
 '0 to 3 Yrs',
 '0 to 1 Yr',
 '0 to 3 Yrs',
 '0 to 3 Yrs',
 '0 to 1 Yr',
 '0 to 4 Yrs',
 '0 to 4 Yrs',
 '0 to 1 Yr']

In [113]:
# Creating a Pandas Dataframe
job_delhi = pd.DataFrame({})
job_delhi["Job Title"] = title_job_name[:10]
job_delhi["Company Name"] = company_name2[:10]
job_delhi["Experience Required"] = exp_required2[:10]
job_delhi["Location"] = job_location2[:10]

In [114]:
job_delhi

Job Title  \
0                           Data Scientist for Delhi   
1                              Data Scientist- Delhi   
2                               Data Scientist-Delhi   
3                           Clinical Data Management   
4                              Data Scientist- Delhi   
5                              Data Scientist- Delhi   
6                              Clinical Data Manager   
7  Work Full/part-time / Work From Home / Compute...   
8  Typing Work-Data Entry-Computer Operator-Freel...   
9                              Clinical Data Analyst   

                   Company Name Experience Required Location  
0                     get hired          0 to 3 Yrs    Delhi  
1             the fashion cosmo          0 to 3 Yrs    Delhi  
2             shiva hr services          0 to 3 Yrs    Delhi  
3                 techno endura           0 to 1 Yr   Delhi6  
4             the fashion cosmo          0 to 3 Yrs    Delhi  
5             the fashion cosmo          0 to 3 Yrs    Delhi  
6                 techno endura           0 to 1 Yr   Delhi6  
7  desnor destination pvt. ltd.          0 to 4 Yrs   Delhi4  
8  desnor destination pvt. ltd.          0 to 4 Yrs   Delhi4  
9                 techno endura           0 to 1 Yr   Delhi6

Q4: Scrape data of first 100 sunglasses listings on flipkart.com. You have to scrape four attributes:

6. Brand

7. Product Description

8. Price

In [116]:
# Opening an automated web browser and website
driver3 = webdriver.Chrome()
driver3.get("https://www.flipkart.com/")

In [134]:
import requests

In [136]:
# Asking for request code from the webpage since it was not allowing to scrape data
page = requests.get("https://www.flipkart.com/")
page

<Response [500]>

Response code [500]> so scraping data from Flipkart.com is not possible

Q5: Scrape 100 reviews data from flipkart.com for iphone11 phone. You have to go the link: https://www.flipkart.com/apple-iphone-11-black-64-gb/product- reviews/itm4e5041ba101fd?pid=MOBFWQ6BXGJCEYNY&lid=LSTMOBFWQ6BXGJCEYNYZXSHRJ&market place=FLIPKART

In [137]:
# Asking for request code from the webpage since it was not allowing to scrape data
url = requests.get("https://www.flipkart.com/search?q=iphone11&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=on&as=off")
url

<Response [500]>

Flipkart.com is sending a response code [500]> making it impossible for data scraping 

Q6: Scrape data for first 100 sneakers you find when you visit flipkart.com and search for “sneakers” in the search field.
You have to scrape 3 attributes of each sneaker:
1. Brand
2. Product Description
3. Price

In [138]:
# Asking for request code from the webpage since it was not allowing to scrape data
url = requests.get("https://www.flipkart.com/")
url

<Response [500]>

Response code [500]>  even after refreshing the page, changing the PHP version and clearing the cache and cookies.

Q7: Go to webpage https://www.amazon.in/ Enter “Laptop” in the search field and then click the search icon. Then set CPU Type filter to “Intel Core i7” as shown in the below image:

In [139]:
# Asking for request code from the webpage since it was not allowing to scrape data
url = requests.get("https://www.amazon.in/s?k=laptop&rh=n%3A1375424031%2Cp_n_feature_thirteen_browse-bin%3A12598163031&dc&ds=v1%3A8WQPYZm5IlAE3VmvJAkMQDzltjUpUhnw2ijTqCU5g4U&crid=2LAE7ZXB27EE2&qid=1701358389&rnid=12598141031&sprefix=laptop%2Caps%2C198&ref=sr_nr_p_n_feature_thirteen_browse-bin_12")
url

<Response [503]>

Response code showing [503]> the server is overloading and not processing my request

 Q8: Write a python program to scrape data for Top 1000 Quotes of All Time.
The above task will be done in following steps:
1. First get the webpagehttps://www.azquotes.com/
2. Click on Top Quotes
3. Than scrap a) Quote b) Author c) Type Of Quotes

In [142]:
# Opening an automated web browser and website
driver = webdriver.Chrome()
driver.get("https://www.azquotes.com/")

In [143]:
# Scraping Data for Quotes from the website
quote_tag = driver.find_elements(By.XPATH,'//a[@class="title"]')
quote_tag

[<selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_167")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_1142")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_166")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2177")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2178")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2179")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4

In [144]:
# Creating an empty list for quotes and looping through them
quote = []
for i in quote_tag:
    quote.append(i.text)
    
    
quote

['The essence of strategy is choosing what not to do.',
 'One cannot and must not try to erase the past merely because it does not fit the present.',
 'Patriotism means to stand by the country. It does not mean to stand by the president.',
 'Death is something inevitable. When a man has done what he considers to be his duty to his people and his country, he can rest in peace. I believe I have made that effort and that is, therefore, why I will sleep for the eternity.',
 'You have to love a nation that celebrates its independence every July 4, not with a parade of guns, tanks, and soldiers who file by the White House in a show of strength and muscle, but with family picnics where kids throw Frisbees, the potato salad gets iffy, and the flies die from happiness. You may think you have overeaten, but it is patriotism.',
 'Be more concerned with your character than your reputation, because your character is what you really are, while your reputation is merely what others think you are.',
 

In [146]:
# Scraping author names from the website
author_tags = driver.find_elements(By.XPATH,'//div[@class="author"]')
author_tags

[<selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_125")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2000")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2289")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2290")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2291")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2292")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c

In [147]:
# Creating an empty list for author names and looping through them
author = []
for i in author_tags:
    author.append(i.text)
    
author

['Michael Porter',
 'Golda Meir',
 'Theodore Roosevelt',
 'Nelson Mandela',
 'Erma Bombeck',
 'John Wooden',
 'Albert Einstein',
 'Frank Zappa',
 'William Faulkner',
 'Will Rogers',
 'Jimmy Carter',
 'Albert Einstein',
 'Abraham Lincoln',
 'Steven Weinberg',
 'Franklin D. Roosevelt',
 'Albert Einstein',
 'Maya Angelou',
 'Mark Twain',
 'Theodore Roosevelt',
 'Michael Jackson',
 'Martin Luther King, Jr.',
 'C. S. Lewis',
 'Theodore Roosevelt',
 'Mark Twain',
 'Abraham Lincoln',
 'Ruth Graham',
 'C. S. Lewis',
 'Groucho Marx',
 'John Lennon',
 'Barbara Bush',
 'Albert Einstein',
 'Francis Chan',
 'Henry Ford',
 'Victor Hugo',
 'Abu Bakr',
 'Zig Ziglar',
 'Winston Churchill',
 'Zig Ziglar',
 'John F. Kennedy',
 'Dwight D. Eisenhower',
 'Winston Churchill',
 'Albert Einstein',
 'Willie Nelson',
 'Johnny Cash',
 'Marie Curie',
 'Maya Angelou',
 'Amy Bloom',
 'Zig Ziglar',
 'John Lennon',
 'Thomas Sowell',
 'Max Lucado',
 'Martin Luther King, Jr.',
 'Thomas Sowell',
 'Salma Hayek',
 'Martin 

In [148]:
# Scraping details of the quote type from the website
type_tags = driver.find_elements(By.XPATH,'//div[@class="tags"]')
type_tags

[<selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2411")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2424")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2425")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2426")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2427")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65c4ee6c658583baff", element="B948502FF85F975270E3DA40752F5145_element_2428")>,
 <selenium.webdriver.remote.webelement.WebElement (session="dec8ca6368245c65

In [149]:
# Creating an empty list for the details of the quote and looping through it
type_of_quote = []
for i in type_tags:
    type_of_quote.append(i.text)
    
type_of_quote

['Essence, Deep Thought, Transcendentalism',
 'Inspiration, Past, Trying',
 'Country, Peace, War',
 'Inspirational, Motivational, Death',
 '4th Of July, Food, Patriotic',
 'Inspirational, Success, Basketball',
 'Strong, Revenge, Intelligent',
 'Inspirational, Teacher, Religious',
 'Truth, Honesty, Lying',
 'Funny, Reading, Learning',
 'Strong, Thoughtful, Compassion',
 'Love, Funny, Life',
 'Freedom, Men, Democracies Have',
 'God, Religious, Atheist',
 'Respect, Kindness, Character',
 'Inspirational, Motivational, Change',
 'Inspirational, Life, Inspiring',
 'Love, Inspirational, Life',
 'Inspiring, Country, 4th Of July',
 'Inspirational, Dream, Hate',
 'Love, Inspirational, Life',
 'Forgiveness, God, Christian',
 'Country, Men, Squares',
 'Inspirational, Relationship, Positive',
 'Inspirational, Motivational, Positive',
 'Love, Friendship, Relationship',
 'Faith, God, Christian',
 'Art, Freedom, Political Will',
 'Inspirational, Life, Faith',
 'Inspirational, Life, Meaningful',
 'Happ

In [150]:
# Creating a Pandas Dataframe for the data
df = pd.DataFrame({})
df["Quote"] = quote
df["Author"] = author
df["Quote Type"] = type_of_quote

In [151]:
df

Quote                Author  \
0   The essence of strategy is choosing what not t...        Michael Porter   
1   One cannot and must not try to erase the past ...            Golda Meir   
2   Patriotism means to stand by the country. It d...    Theodore Roosevelt   
3   Death is something inevitable. When a man has ...        Nelson Mandela   
4   You have to love a nation that celebrates its ...          Erma Bombeck   
..                                                ...                   ...   
95     When the going gets weird, the weird turn pro.    Hunter S. Thompson   
96  When a train goes through a tunnel and it gets...       Corrie Ten Boom   
97  If you think you are too small to make a diffe...            Dalai Lama   
98  God doesn't require us to succeed, he only req...         Mother Teresa   
99    Change your thoughts and you change your world.  Norman Vincent Peale   

                                  Quote Type  
0   Essence, Deep Thought, Transcendentalism  
1                  Inspiration, Past, Trying  
2                        Country, Peace, War  
3         Inspirational, Motivational, Death  
4               4th Of July, Food, Patriotic  
..                                       ...  
95                    Music, Sports, Hunting  
96             Trust, Encouraging, Uplifting  
97              Inspirational, Funny, Change  
98                      Success, God, Mother  
99       Inspirational, Motivational, Change  

[100 rows x 3 columns]

Q9: Write a python program to display list of respected former Prime Ministers of India(i.e. Name, Born-Dead, Term of office, Remarks) from https://www.jagranjosh.com/.
This task will be done in following steps:
1. First get the webpagehttps://www.jagranjosh.com/
2. Then You have to click on the GK option
3. Then click on the List of all Prime Ministers of India
4. Then scrap the mentioned data and make theDataFrame.

In [152]:
driver = webdriver.Chrome()
driver.get("https://www.jagranjosh.com/")

In [153]:
name_tags = driver.find_elements(By.XPATH,'//a[@target="_blank"]')
name_tags

[<selenium.webdriver.remote.webelement.WebElement (session="b721292fd6f255bb46386fbb4d285494", element="5533F75D25240F063C84131DBFD43E52_element_1026")>,
 <selenium.webdriver.remote.webelement.WebElement (session="b721292fd6f255bb46386fbb4d285494", element="5533F75D25240F063C84131DBFD43E52_element_1027")>,
 <selenium.webdriver.remote.webelement.WebElement (session="b721292fd6f255bb46386fbb4d285494", element="5533F75D25240F063C84131DBFD43E52_element_1028")>,
 <selenium.webdriver.remote.webelement.WebElement (session="b721292fd6f255bb46386fbb4d285494", element="5533F75D25240F063C84131DBFD43E52_element_1029")>,
 <selenium.webdriver.remote.webelement.WebElement (session="b721292fd6f255bb46386fbb4d285494", element="5533F75D25240F063C84131DBFD43E52_element_1030")>,
 <selenium.webdriver.remote.webelement.WebElement (session="b721292fd6f255bb46386fbb4d285494", element="5533F75D25240F063C84131DBFD43E52_element_1031")>,
 <selenium.webdriver.remote.webelement.WebElement (session="b721292fd6f255bb

In [154]:
name = []
for i in name_tags:
    name.append(i.text)
    
name

['SRM UNIVERSITY',
 'NIKHARDA PUNJAB',
 '',
 '',
 '',
 '',
 'Jawahar Lal Nehru',
 'Indira Gandhi',
 'Indira Gandhi',
 'Rajiv Gandhi',
 'Chandra Shekhar',
 'Atal Bihari Vajpayee',
 'Atal Bihari Vajpayee',
 'Manmohan Singh',
 'Narendra Modi',
 'Prime Minister',
 'President',
 'List of all Presidents of India',
 'List of Nicknames of Indian Prime Ministers',
 '',
 '',
 '',
 '',
 '',
 '']

In [155]:
life_tags = driver.find_elements(By.XPATH,'//p[@class="empty"]')
life_tags

[]

In [156]:
life = []
for i in life_tags:
    life.append(i.text)

In [157]:
office_tags = driver.find_elements(By.XPATH,'//span[@class="empty"]')

In [ ]:
office = []
for i in office_tags:
    office.append(i.text)

In [158]:
remarks_tags = driver.find_elements(By.XPATH,'//p[@class="empty"]')

In [ ]:
remarks = []
for i in remarks_tags:
    remarks.append(i.text)

In [ ]:
df1 = pd.DataFrame({})
df1["Name"] = name
df1["Born and Death"] = life
df1["Term of Office"] = office
df1["Remarks"] = remarks

Q10: Write a python program to display list of 50 Most expensive cars in the world (i.e. Car name and Price) from https://www.motor1.com/
This task will be done in following steps:
1. First get the webpage https://www.motor1.com/
2. Then You have to type in the search bar ’50 most expensive cars’
3. Then click on 50 most expensive cars in the world..
4. Then scrap the mentioned data and make the dataframe.

In [159]:
driver = webdriver.Chrome()
driver.get("https://www.motor1.com/")

In [160]:
name_tags_car = driver.find_elements(By.XPATH,'//h3[@class="subheader"]')
name_tags_car

[<selenium.webdriver.remote.webelement.WebElement (session="fbbd882bdf0ba1f09e7abe7dc96fee58", element="08A583417AC58CAC922B34104E81A1C9_element_128")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fbbd882bdf0ba1f09e7abe7dc96fee58", element="08A583417AC58CAC922B34104E81A1C9_element_392")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fbbd882bdf0ba1f09e7abe7dc96fee58", element="08A583417AC58CAC922B34104E81A1C9_element_396")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fbbd882bdf0ba1f09e7abe7dc96fee58", element="08A583417AC58CAC922B34104E81A1C9_element_402")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fbbd882bdf0ba1f09e7abe7dc96fee58", element="08A583417AC58CAC922B34104E81A1C9_element_407")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fbbd882bdf0ba1f09e7abe7dc96fee58", element="08A583417AC58CAC922B34104E81A1C9_element_412")>,
 <selenium.webdriver.remote.webelement.WebElement (session="fbbd882bdf0ba1f09e7abe

In [161]:
name = []
for i in name_tags_car:
    name.append(i.text)
    
name

['Aston Martin Valour',
 'McLaren Elva',
 'Czinger 21C',
 'Ferrari Monza',
 'Gordon Murray T.33',
 'Koenigsegg Gemera',
 'Zenvo TSR-S',
 'Hennessey Venom F5',
 'Bentley Bacalar',
 'Hispano Suiza Carmen Boulogne',
 'Bentley Mulliner Batur',
 'Deus Vayanne',
 'SSC Tuatara',
 'Lotus Evija',
 'Aston Martin Vulcan',
 'Delage D12',
 'Ferrari Daytona SP3',
 'McLaren Speedtail',
 'Rimac Nevera',
 'Pagani Utopia',
 'Pininfarina Battista',
 'Gordon Murray T.50',
 'Lamborghini Countach',
 'Mercedes-AMG Project One',
 'Zenvo Aurora',
 'Aston Martin Victor',
 'Hennessey Venom F5 Roadster',
 'Koenigsegg Jesko',
 'Aston Martin Valkyrie',
 'W Motors Lykan Hypersport',
 'McLaren Solus',
 'Lamborghini Sian',
 'Koenigsegg CC850',
 'Bugatti Chiron Super Sport 300+',
 'Lamborghini Veneno',
 'Bugatti Bolide',
 'Pininfarina B95 Speedster',
 'Bugatti Mistral',
 'Pagani Huayra Imola',
 'Bugatti Divo',
 'SP Automotive Chaos',
 'Pagani Codalunga',
 '777 Hypercar',
 'Mercedes-Maybach Exelero',
 'Bugatti Centodiec

In [162]:
price_tags = driver.find_elements(By.XPATH,'//strong[@class="empty"]')

In [163]:
price = []
for i in price_tags:
    price.append(i.text)

In [ ]:
df_cars = pd.DataFrame({})
df_cars["Name of Car"] = name
df["Price of Car"] = price